Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h

```diff
@@ -480,18 +483,35 @@ void LoadEventNexus::loadEntryMetadata(const std::string &nexusfilename, T WS,
     // let it drop on floor
   }
 
-  // get the sample name
+  // get the sample name - nested try/catch to leave the handle in an
+  // appropriate state
   try {
     file.openGroup("sample", "NXsample");
-    file.openData("name");
-    std::string name;
-    if (file.getInfo().type == ::NeXus::CHAR) {
-      name = file.getStrData();
-    }
-    if (!name.empty()) {
-      WS->mutableSample().setName(name);
+    try {
+      file.openData("name");
+      const auto info = file.getInfo();
+      std::string name;
+      if (info.type == ::NeXus::CHAR) {
+        if (info.dims.size() == 1) {
+          name = file.getStrData();
+        } else { // something special for 2-d array
+          const int64_t total_length =
+              std::accumulate(info.dims.begin(), info.dims.end(), 1,
+                              std::multiplies<int64_t>());
+          boost::scoped_array<char> val_array(new char[total_length]);
```

Comment:
Using `boost::scoped_array<char>` works, though `std::vector<char>` may be simpler and carry more information in the object.